src/Illuminate/Support/Str.php

```diff
@@ -419,6 +419,45 @@ public static function isUuid($value)
         return preg_match('/^[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}$/iD', $value) > 0;
     }
 
+    /**
+     * Join items into a human readable list (e.g. "one, two, three, and four")
+     * Uses different glue strings when there are only two elements and for
+     * the final element. Defaults to joining using the Oxford comma.
+     *
+     * 1 item will return: $item
+     * 2 items will return: $item1 . $dyadicGlue . $item2
+     * 3+ items will return: $item1 . $glue . $item2 . $lastGlue . $item3
+     */
+    public static function join(iterable $items, string $glue = ', ', string $lastGlue = ', and ', $dyadicGlue = ' and '): string
```

Comment:
- Code doc `@params` and `@returns` annotations are missing
- A return type is given (`:string`). No other methods in the `Str` class have a return type.

Reply:
Code docs are redundant when proper type hints are already provided on the parameters and for the return type, all they do open the possibility that they will become out of date if future changes are made; thus they have been intentionally excluded in favour of the language level type hinting.

Just because none of the other methods have a return type set yet doesn't mean that new ones can't or shouldn't (they should IMHO). 

The name is actually the best choice here, given that Arr::join() and Collection::join() both exist already and provide their parameters in the same order as this method while also performing similar functionality; this method just also provides the $diadicGlue parameter as well. `listify()` doesn't make much sense to me personally.

Reply:
Okay, but your PR will be rejected for not complying with Laravel’s style rules.

Reply:
The [style guide](https://laravel.com/docs/9.x/contributions#coding-style) does not make a statement one way or the other whether you must include PHPDoc comments at all times, even when they're redundant; but I appreciate your concern. Do you have a more specific reference where they are listed as required?

Reply:
@LukeTowers everywhere else in the code. Please remove the scalar types and format the docblocks like in the rest of the framework.